webpack/assets/javascripts/react_app/components/HostDetails/DetailsCard/PowerStatus/PowerStatusDropDown.js

```diff
@@ -0,0 +1,90 @@
+import PropTypes from 'prop-types';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  Dropdown,
+  DropdownToggle,
+  DropdownItem,
+  Tooltip,
+} from '@patternfly/react-core';
+
+import { foremanUrl } from '../../../../common/helpers';
+import { useAPI } from '../../../../common/hooks/API/APIHooks';
+import { translate as __ } from '../../../../common/I18n';
+import PowerStatusIcon from './PowerStatusIcon';
+import {
+  POWER_REQUEST_OPTIONS,
+  BASE_POWER_STATES,
+  SUPPORTED_POWER_STATES,
+} from './constants';
+import { changeHostPower } from './actions';
+import '../styles.scss';
+
+const PowerStatusDropDown = ({ hostID, hasPowerPermission, isBmc }) => {
+  const powerURL = foremanUrl(`/api/hosts/${hostID}/power`);
+  const {
+    response: { state: currentState, title, statusText },
+    status: responseStatus,
+  } = useAPI('get', powerURL, POWER_REQUEST_OPTIONS);
+
+  const dispatch = useDispatch();
+  const [isOpen, setOpen] = useState(false);
+  const changePowerHandler = targetState => {
+    // TODO: replace with confirmation service
+    window.confirm(__('Are you sure?')) &&
+      dispatch(changeHostPower(targetState, hostID));
+  };
+  const dropdownItems = () => {
+    const supportedPowerStates = isBmc
+      ? SUPPORTED_POWER_STATES
+      : BASE_POWER_STATES;
+
+    return Object.keys(supportedPowerStates).map(state => (
+      <DropdownItem
+        onClick={() => changePowerHandler(state)}
+        isDisabled={state === currentState}
+        key={state}
+      >
+        {supportedPowerStates[state]}
+      </DropdownItem>
+    ));
+  };
+
+  const onDropdownSelect = event => setOpen(false);
+  const onToggle = open => setOpen(open);
+  return (
+    <Tooltip content={statusText || title}>
```

Comment:
When there is an error in the api call `statusText` and `title` are undefined. This causes an empty tooltip.
Should the dropdown be shown if there is an error?
![Screenshot from 2021-09-07 20-36-28](https://user-images.githubusercontent.com/30431079/132394378-a1a4ba44-3c4c-49f8-8ceb-9079f9910565.png)

Reply:
if there is any error during fetching the API call, [statusText](https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/hosts_controller.rb#L281) should contain the error, what error do you get from the server?

Reply:
I see, this was probably a fake api error. So no issue here